src/components/index/IndexComponent.tsx

```diff
@@ -21,13 +79,89 @@ const Index = () => {
           <HeaderTitleText> Requests &amp; Proposals </HeaderTitleText>{" "}
         </HeaderTitle>
       </Header>
+      <FilterWrapper>
+        <FilterButtonRow>
+          <FilterButton
+            variant={filter === Filter.DEFAULT ? "primary" : "outline"}
+            onClick={() => setFilter(Filter.DEFAULT)}
+          >
+            <div>All</div>{" "}
+            <FilterNumbers selected={filter === Filter.DEFAULT}>
+              {numAll}
+            </FilterNumbers>
+          </FilterButton>
+          <FilterButton
+            variant={filter === Filter.REQUESTS ? "primary" : "outline"}
+            onClick={() => setFilter(Filter.REQUESTS)}
+          >
+            <div>Requests </div>{" "}
+            <FilterNumbers selected={filter === Filter.REQUESTS}>
+              {numRequested}
+            </FilterNumbers>
+          </FilterButton>
+          <FilterButton
+            variant={filter === Filter.PROPOSED ? "primary" : "outline"}
+            onClick={() => setFilter(Filter.PROPOSED)}
+          >
+            <div>Proposed </div>{" "}
+            <FilterNumbers selected={filter === Filter.PROPOSED}>
+              {numProposed}
+            </FilterNumbers>
+          </FilterButton>
+          <FilterButton
+            variant={filter === Filter.DISPUTED ? "primary" : "outline"}
+            onClick={() => setFilter(Filter.DISPUTED)}
+          >
+            <div>Disputed </div>{" "}
+            <FilterNumbers selected={filter === Filter.DISPUTED}>
+              {numDisputed}
+            </FilterNumbers>
+          </FilterButton>
+          <FilterButton
+            onClick={() => setChecked((prevValue) => !prevValue)}
+            variant="base"
+          >
+            <input
+              checked={checked}
+              type="checkbox"
+              onChange={(event) => setChecked(event.target.checked)}
+            />
+            <div>Show Answered</div>
+            <div>{numAnswered}</div>
+          </FilterButton>
+        </FilterButtonRow>
+      </FilterWrapper>
       <Body>
         <TableRow>
-          <RequestsTable />
+          <RequestsTable requests={filteredRequests} />
         </TableRow>
       </Body>
     </Wrapper>
   );
 };
 
+function ALL_FILTER(x: RequestIndex) {
+  return (
+    x.state === RequestState.Requested ||
+    x.state === RequestState.Proposed ||
+    x.state === RequestState.Disputed
```

Comment:
this should be a composition of other filters, so you reduce the places you have to change something

```js
   return ( REQUEST_FILTER(x) || PROPOSED_FILTER(x) || ..etc
```

Reply:
Fixed.